### PR TITLE
CTCTRADERS-1094: Add CustomsOffice models (departure, destination, transit)

### PR DESCRIPTION
**Pull request by chris-ianson**

### CODE REVIEW

File: app/models/messages/customsoffice/CustomsOffice.scala [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.messages.customsoffice
+
+object CustomsOffice {
+  val length              = 17
+  val transitOfficeCount = 9
```
Comment (amaalali) [inline app/models/messages/customsoffice/CustomsOffice.scala:21]: For consistency should we nest in a `Constants` object?

File: app/models/messages/customsoffice/CustomsOfficeTransit.scala [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  implicit def writes: XMLWrites[CustomsOfficeTransit] = XMLWrites[CustomsOfficeTransit] {
+    customsOffice =>
+      val arrivalTime = customsOffice.arrivalTime.map {
+        arrivalTime =>
+          <ArrTimTRACUS085>{Format.dateTimeFormattedIE015(arrivalTime)}</ArrTimTRACUS085>
+      }
+
+      <CUSOFFTRARNS>
+        <RefNumRNS1>{customsOffice.referenceNumber}</RefNumRNS1>
+        {arrivalTime.getOrElse(NodeSeq.Empty)}
```
Comment (amaalali) [inline app/models/messages/customsoffice/CustomsOfficeTransit.scala:47]: This pattern of optional writes is showing up a lot, I think there is a missing abstraction 🤔  We can change `XMLWrites` to make this easier

**Review by amaalali — APPROVED** [review]

File: app/models/messages/customsoffice/CustomsOffice.scala (hunk 1 above)
Comment (chris-ianson) [reply app/models/messages/customsoffice/CustomsOffice.scala:21]: This object is only ever going to contain length values but I can wrap it in Constants to keep it consistent with the others 👍

File: app/models/messages/customsoffice/CustomsOfficeTransit.scala (hunk 2 above)
Comment (chris-ianson) [reply app/models/messages/customsoffice/CustomsOfficeTransit.scala:47]: I'll have a look at this and will probably give you a shout to go over @amaalali

**Pull request merged**